### PR TITLE
Models

### PR DESCRIPTION
**Pull request by Adel-Wehbi**
Most of the initial work for the models, and more unit tests than is probably necessary.

The GameModel can now initialize by placing the snake in its initial position. I also added the logic to move the snake in any direction, with a variable to control whether the snake grows by one during the movement.

Unit tests can be run like this: `python tests.py -v`, where python is version 3 or above.

### CODE REVIEW

**Review by jaredlueck — APPROVED** [review]

File: SnakeGame/model.py [hunk 1]
```diff
@@ -22,12 +10,113 @@ def __init__(self):
         self.initSnakeLen = self.DEFAULT_INIT_SNAKE_LEN
 
     def init(self):
-        # start with an empty grid
-        self.grid = [list([self.CellContent.Empty] * self.gridSize[0]) for x in range(self.gridSize[1])]
```
Comment (Adel-Wehbi) [inline SnakeGame/model.py:26]: Seeing as we'll have to keep a reference of the position of the snake independent of the Grid, I decided to forego this implementation in favor of generating the grid when needed (that is to say: at the end of game tick, after all logic has completed executing).

File: SnakeGame/model.py [hunk 2]
```diff
@@ -1,18 +1,6 @@
 import enum
 
-class GameModel(object):
-
-    class CellContent(enum.Enum):
```
Comment (eleanorRumsey) [inline SnakeGame/model.py:5]: Why were these deleted?

Comment (Adel-Wehbi) [reply SnakeGame/model.py:5]: They weren't deleted. They were moved outside the GameModel class. The reason for that is, if we were to import those enums in the other files (as we will), the previous way would have been: `from model import GameModel`. We would have had to reference the enums as `GameModel.CellContent`. That would be cumbersome. 

The way it's done now would be like: `from model import CellContent`

File: SnakeGame/model.py [hunk 3]
```diff
@@ -22,12 +10,113 @@ def __init__(self):
...
+
+    def moveSnake(self, growSnake=False):
+        '''Moves the snake in the Direction set in the object.
+        If grow is True, the snake will grow by one block while moving.'''
+        gridWidth = self.gridSize[0]
+        gridLength = self.gridSize[1]
+        # we only need to determine where the head goes next
+        # and then all the other blocks switch places with their previousBlock each
+        head = self.snake.body[0]
+        previousBlock = head.copy()
```
Comment (eleanorRumsey) [inline SnakeGame/model.py:51]: Unfamiliar with .copy() ... assuming method for reference in python :)

Comment (Adel-Wehbi) [reply SnakeGame/model.py:51]: More or less... Python treats everything as objects, so we need to make a shallow copy of the list to make the information does not get overwritten in the lines below.

**Review by eleanorRumsey — APPROVED** [review]
